cmd/snap/cmd_run.go

```diff
@@ -202,6 +203,16 @@ func (x *cmdRun) Execute(args []string) error {
 	snapApp := args[0]
 	args = args[1:]
 
+	// If the snap is inhibited from being used then postpone running it until
+	// that condition passes. Inhibition UI can be dismissed by the user, in
+	// which case we don't run the application at all.
+	if features.RefreshAppAwareness.IsEnabled() {
+		snapName, _ := snap.SplitSnapApp(snapApp)
+		if err := waitWhileInhibited(snapName); err != nil {
```

Comment:
I think we shouldn't do this if the app is a service I think, we have other mechanisms to deal with that, and I fear this code might end up into confusing situation with services

Reply:
Do you mean we should just not implement inhibition at all?

Reply:
we can discuss, the problem is that this arbitrary waiting if a service gets this far will mess up with systemd timeouts and notion of failures. We have code to disable services while we refresh already, or not disable them if the snap ask us not to

Reply:
I agree with the rationale, I'm almost done with the other, related, branch and I should be able to address all the feedback tomorrow.

Reply:
I've addressed this point.